constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/Transition.java

```diff
@@ -274,26 +281,42 @@ float getDestinationPosition(float currentPosition, float velocity, float durati
                     return 1;
                 case ON_UP_STOP:
                     return Float.NaN;
-                case ON_UP_AUTOCOMPLETE:
                 case ON_UP_DECELERATE:
+                    return Math.max(0, Math.min(1, rest));
                 case ON_UP_DECELERATE_AND_COMPLETE:
+                   if (rest > 0.2f && rest < 0.8f){
```

Comment:
What's the reason to choose 0.2 as the threshold?

Reply:
Just random for now. 
The concept is decelerate just slows down.
decelerate & complete can complete if close.
The problem is it is meaning less if it is impossible if two difficult to complete.
So if you were sliding on a 5 inch slider if it gets 0-1 or 4-5 inch it completes. 
What happens typically is developer just listens to a decelerate and codes there own complete.  (clock approach)